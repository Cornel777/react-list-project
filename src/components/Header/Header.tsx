import { FunctionComponent } from 'react';
import HeaderContainer from './styled/HeaderContainer';

interface HeaderProps {
  children: React.ReactNode;
}

const Header: FunctionComponent<HeaderProps> = ({children}) => {
  return (
    <HeaderContainer>
      <img 
        src='https://images.seeklogo.com/logo-png/13/1/star-wars-logo-png_seeklogo-131743.png' 
        loading='lazy' 
        alt='Star Wars Logo'
      >
      </img>
      {children}
    </HeaderContainer>
  );
}

export default Header;
