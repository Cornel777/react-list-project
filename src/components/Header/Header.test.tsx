import { render } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import Header from './Header';
import '@testing-library/jest-dom';

describe('Header component', () => {
    it('should match the snapshot', () => {
        const { asFragment } = render(
          <Header>
            <nav>Test Content</nav>
          </Header>
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render the Star Wars logo', () => {
        const { getByAltText } = render(
          <Header>
            <nav>Test Content</nav>
          </Header>
        );
        const logo = getByAltText('Star Wars Logo');
        expect(logo).toBeInTheDocument();
    });
});