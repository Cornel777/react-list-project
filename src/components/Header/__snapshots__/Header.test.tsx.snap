// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header component > should match the snapshot 1`] = `
<DocumentFragment>
  <header
    class="sc-KXCwU cLCwkl"
  >
    <img
      alt="Star Wars Logo"
      loading="lazy"
      src="https://images.seeklogo.com/logo-png/13/1/star-wars-logo-png_seeklogo-131743.png"
    />
    <nav>
      Test Content
    </nav>
  </header>
</DocumentFragment>
`;
